src/app/hero/hero-data.service.ts

```diff
@@ -1,9 +1,34 @@
-import { Injectable } from '@angular/core';
+import {Injectable} from '@angular/core';
+import {HttpClient} from "@angular/common/http";
+import {Observable} from "rxjs";
+import {map, tap} from "rxjs/operators";
+import {Hero} from "../model/hero";
+import {environment} from "../../environments/environment";
 
 @Injectable({
   providedIn: 'root'
 })
 export class HeroDataService {
 
-  constructor() { }
+  constructor(private http: HttpClient) {
+  }
+
+  getHeroDetail(heroId: number): Observable<Hero> {
+    return this.http
+      .get<any>(environment.getCharacterUrl+heroId+"?apikey="+environment.apiKey)
+      .pipe(
+        map(
+          rawData => <Hero> {
```

Comment:
instead you can map the 0th array directly 
rawData => rawData .data.results[0]

and send the all the details for now. since this api call might be used by others as well, we cannot be specific about what data to map.
you can filter out the mappings when you subscribe to the api